go/libraries/doltcore/sqle/index_lookup.go

```diff
@@ -29,20 +31,116 @@ type IndexLookupKeyIterator interface {
 }
 
 type doltIndexLookup struct {
-	idx     DoltIndex
-	keyIter IndexLookupKeyIterator
+	idx       DoltIndex
+	lookupOps []lookupops.LookupOp
 }
 
+var _ sql.MergeableIndexLookup = (*doltIndexLookup)(nil)
+
 func (il *doltIndexLookup) String() string {
 	// TODO: this could be expanded with additional info (like the expression used to create the index lookup)
 	return fmt.Sprintf("doltIndexLookup:%s", il.idx.ID())
 }
 
-var _ sql.IndexLookup = (*doltIndexLookup)(nil)
+// IsMergeable implements sql.MergeableIndexLookup
+func (il *doltIndexLookup) IsMergeable(indexLookup sql.IndexLookup) bool {
+	otherIl, ok := indexLookup.(*doltIndexLookup)
+	if !ok {
+		return false
+	}
+	return il.idx == otherIl.idx
+}
+
+// Intersection implements sql.MergeableIndexLookup
+//TODO: change signature to accept errors
```

Comment:
Another approach is to return an allows-erroring-IndexLookup implementation...

Reply:
The interface was updated altogether (breaking change) in `go-mysql-server` already to allow for errors 😄